projects/storefrontlib/cms-structure/guards/multi-cms-page-guard.service.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { CX_PAGE_GUARD_TOKEN } from './cms-page-guard.provider';
+import { RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable, concat, endWith, first, of, skipWhile } from 'rxjs';
+import { CmsGuardsService } from '../services';
+import { CmsActivatedRouteSnapshot, UnifiedInjector } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MultiCmsPageGuardService {
+  protected cxPageGuards = inject(CX_PAGE_GUARD_TOKEN);
+  protected cmsGuardService = inject(CmsGuardsService);
```

Comment:
IMHO we should not depend here the `CmsGuardsService` (which responsibility is to resolve the **cms-mapping**-driven guards).

Instead, I suggest to extract most of the logic from the existing `CmsGuardsService` to a new service like `ManyGuardsService` with 2 methods:
```ts
  canActivateGuards(
    guards: CanActivate[], 
    route: CmsActivatedRouteSnapshot,
    state: RouterStateSnapshot
  }) { ... }

  canActivateGuard(
    guards: CanActivate, 
    route: CmsActivatedRouteSnapshot,
    state: RouterStateSnapshot
  }) { ... }
```

Then we could reuse this new service in the internals of the existing `CmsGuardsService` and also reuse this new service here.

In other words, we only need here an util service for handling many guards, but we don't need resolving **cms-mapping**-driven guards

Reply:
UPDATE: I've noticed only now that the existing CmsPageService uses sophisticated mechanisms like UnifiedInjector, that I believe are not needed here. Therefore the amount of code that could be reuse is much smaller than I thought. I'm sorry, bear with me... I need to think about it more and I'll get back to you...

Reply:
done. proposed changes in another PR do develop https://github.com/SAP/spartacus/pull/18781